File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3-coder-30b-mlx_nothink/20260830_231050/typescript/002_word_frequency/solution.ts
import fs from 'fs';
import path from 'path';

// Read the file content
const filePath = path.join('input', 'text.txt');
const content = fs.readFileSync(filePath, 'utf8');

// Process the content
const words = content
  .toLowerCase()
  .replace(/[^a-z\s]/g, ' ')
  .split(/\s+/)
  .filter(word => word.length > 0);

// Count frequencies
const wordCount: Record<string, number> = {};
for (const word of words) {
  wordCount[word] = (wordCount[word] || 0) + 1;
}

// Sort and output
const sortedEntries = Object.entries(wordCount)
  .sort(([wordA, countA], [wordB, countB]) => {
    if (countB !== countA) {
      return countB - countA; // Descending by count
    }
    return wordA.localeCompare(wordB); // Ascending by word
  });

for (const [word, count] of sortedEntries) {
  console.log(`${word}: ${count}`);
}

File: codegen/results/v002/m3-max-48gb/lmstudio/qwen3-coder-30b-mlx_nothink/20260830_231050/go/002_word_frequency/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strings"
	"unicode"
)

func main() {
	file, err := os.Open("input/text.txt")
	if err != nil {
		return
	}
	defer file.Close()

	wordCounts := make(map[string]int)

	scanner := bufio.NewScanner(file)
	for scanner.Scan() {
		line := scanner.Text()
		// Process each line
		words := strings.FieldsFunc(line, func(r rune) bool {
			return !unicode.IsLetter(r)
		})
		
		for _, word := range words {
			// Convert to lowercase
			lowerWord := strings.ToLower(word)
			// Remove any remaining punctuation
			cleanWord := strings.TrimFunc(lowerWord, func(r rune) bool {
				return !unicode.IsLetter(r)
			})
			
			if cleanWord != "" {
				wordCounts[cleanWord]++
			}
		}
	}

	// Convert map to slice for sorting
	type wordCount struct {
		word  string
		count int
	}
	
	var words []wordCount
	for word, count := range wordCounts {
		words = append(words, wordCount{word, count})
	}

	// Sort by count descending, then by word ascending
	sort.Slice(words, func(i, j int) bool {
		if words[i].count == words[j].count {
			return words[i].word < words[j].word
		}
		return words[i].count > words[j].count
	})

	// Output results
	for _, wc := range words {
		fmt.Printf("%s: %d\n", wc.word, wc.count)
	}
}